### PR TITLE
VIM-6641: Adds support for Albums and VIMVideos labeled as Stock

### PR DESCRIPTION
**Pull request by jasonhawkins**
### Ticket
[VIM-6641](https://vimean.atlassian.net/browse/VIm-6641)

### Ticket Summary
Introduce support for requesting and parsing `Album` objects, including password-protected albums. 

Add support for identifying a `VIMVideo` object as a stock video. More information about Vimeo Stock can be found at [vimeo.com/stock](https://vimeo.com/stock).

### Implementation Summary
This pull request merges previously reviewed work into VimeoNetworking's `develop` branch.
- [VIM-6642: Adds support for Albums](https://github.com/vimeo/VimeoNetworking/pull/278)
- [VIM-6642: Creates generalized method for creating password-protected requests](https://github.com/vimeo/VimeoNetworking/pull/303)
- [VIM-6849: Adds support for identifying stock videos](https://github.com/vimeo/VimeoNetworking/pull/304)

### How to Test
- Ensure the sample project builds and runs.
- Ensure unit tests continue to pass.

### CODE REVIEW

File: VimeoNetworking/Sources/Models/Album.swift [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    @objc public var uri: String?
+    @objc public var link: String?
+    @objc public var embed: AlbumEmbed?
+    @objc public var videoThumbnails: NSArray?
+    @objc public var user: VIMUser?
+    @objc public var theme: String?
+    @objc dynamic private var metadata: [String: Any]?
+    @objc dynamic private var connections: [String: Any]?
+    
+    public override func getObjectMapping() -> Any! {
```
Comment (nicolelehrer) [inline VimeoNetworking/Sources/Models/Album.swift]: Does this need to be force unwrapped?

File: VimeoNetworking/Sources/Models/Album.swift [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            Constant.Key.Name : Constant.Value.Name,
+            Constant.Key.Description : Constant.Value.Description,
+            Constant.Key.Logo : Constant.Value.Logo,
+            Constant.Key.CreatedTime : Constant.Value.CreatedTime,
+            Constant.Key.ModifiedTime : Constant.Value.ModifiedTime,
+            Constant.Key.Pictures : Constant.Value.VideoThumbnails
+        ]
+    }
+    
+    public override func getClassForObjectKey(_ key: String!) -> AnyClass? {
```
Comment (nicolelehrer) [inline VimeoNetworking/Sources/Models/Album.swift]: Force unwrapping needed?

File: VimeoNetworking/Sources/Models/Album.swift [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            return nil
+        }
+    }
+    
+    public override func getClassForCollectionKey(_ key: String!) -> AnyClass! {
+        switch key {
+        case Constant.Collections.Pictures:
+            return Constant.Class.PictureCollection
+        default:
+            return nil
```
Comment (nicolelehrer) [inline VimeoNetworking/Sources/Models/Album.swift]: Looks like if we can return nil should this be an optional return value?
Force unwrapping needed on input param?

**Comment by jasonhawkins** [comment]
Thanks for calling those out @nicolelehrer. This is ready for another look.

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          8 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L181">Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L181</a></strong>: ‘setResumingHandler:’ is only available on iOS 9.0 or newer [-Wunguarded-availability]  <br /><code>
        [self.downloadProgress setResumingHandler:^{
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L95</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L104</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L76</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75">Pods/AFNetworking/UIKit+AFNetworking/UIRefreshControl+AFNetworking.m#L75</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L106">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L106</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L163">Pods/AFNetworking/AFNetworking/AFURLSessionManager.m#L163</a></strong>: ‘setResumingHandler:’ is only available on iOS 9.0 or newer [-Wunguarded-availability]  <br /><code>
        [self.uploadProgress setResumingHandler:^{
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoNetworking/blob/18a00aa4930a54fb2ddabf62e08bdd01d4ede538/Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97">Pods/AFNetworking/UIKit+AFNetworking/UIActivityIndicatorView+AFNetworking.m#L97</a></strong>: unknown warning group ‘-Wreceiver-is-weak’, ignored [-Wunknown-warning-option]  <br /><code>
#pragma clang diagnostic ignored "-Wreceiver-is-weak"
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 114 tests, with 0 failures (0 unexpected) in 1.446 (1.704) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoNetworking/984)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: VimeoNetworking/Sources/Models/Album.swift [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        case Constant.Key.Embed:
+            return Constant.Class.Embed
+        case Constant.Key.Metadata:
+            return Constant.Class.MutableDictionary
+        default:
+            return nil
+        }
+    }
+    
+    public override func getClassForCollectionKey(_ key: String?) -> AnyClass! {
```
Comment (nicolelehrer) [inline VimeoNetworking/Sources/Models/Album.swift]: Looks like there's one more here @jasonhawkins

Comment (jasonhawkins) [reply VimeoNetworking/Sources/Models/Album.swift]: ahh sorry about that.

**Pull request merged**